test/conformance/event/urEventRetain.cpp

```diff
@@ -1,2 +1,38 @@
 // Copyright (C) 2023 Intel Corporation
 // SPDX-License-Identifier: MIT
+
+#include "fixtures.h"
+
+using urEventRetainTest = uur::event::urEventReferenceTest;
+
+/* Check that urEventRetain returns Success */
+TEST_P(urEventRetainTest, Success) {
+    ASSERT_NO_FATAL_FAILURE(CreateEvent());
+    ASSERT_SUCCESS(urEventRetain(event));
+    ASSERT_SUCCESS(urEventRelease(event));
+    ASSERT_SUCCESS(urEventRelease(event));
+    EXPECT_NO_FATAL_FAILURE(Cleanup());
+}
+
+/* Check that urEventRetain increments the reference count */
+TEST_P(urEventRetainTest, CheckReferenceCount) {
+
+    ASSERT_NO_FATAL_FAILURE(CreateEvent());
+    ASSERT_NO_FATAL_FAILURE(checkEventReferenceCount(1));
+    ASSERT_SUCCESS(urEventRetain(event));
+    ASSERT_NO_FATAL_FAILURE(checkEventReferenceCount(2));
+    ASSERT_SUCCESS(urEventRelease(event));
+    ASSERT_SUCCESS(urEventRelease(event));
+    EXPECT_NO_FATAL_FAILURE(Cleanup());
+}
+
+TEST_P(urEventRetainTest, InvalidNullHandle) {
+    ASSERT_EQ(urEventRetain(nullptr), UR_RESULT_ERROR_INVALID_NULL_HANDLE);
+}
+
+TEST_P(urEventRetainTest, InvalidEvent) {
+    ur_event_handle_t event{};
+    ASSERT_EQ(urEventRetain(event), UR_RESULT_ERROR_INVALID_EVENT);
```

Comment:
Would this be the correct way to test UR_RESULT_ERROR_INVALID_EVENT?

Reply:
A handle is just a pointer so just give it some random invalid address that isn't null.

```cpp
ur_event_handle_t event = 0xdeadbeef;
```

Reply:
How would an implementation verify that a pointer is pointing to a random place in memory? I think that would lead to segmentation faults. Unless we keep a list of valid pointers internally. Not sure if that's something we would like to do.

Reply:
In UnitUR there was a test for memory that used this https://github.com/oneapi-src/unified-runtime/pull/233/files#diff-8bcf0034cb139b390002ba870b39e7dd2fb245f47d618c02ad74914725b937fdR73. However for allocations we do track valid pointers so can check if a pointer is valid. I don't know if we mandate the same for events

Reply:
Those tests are testing something different @veselypeta, `UR_RESULT_ERROR_INVALID_NULL_HANDLE` rather than `UR_RESULT_ERROR_INVALID_<OBJECT>`, or a fill patten for a setting memory.

I don't think we can reliable trigger `UR_RESULT_ERROR_INVALID_EVENT` without making assumptions about the implementation of an adapter.